### ISSUE TITLE
Issue with community.list

### ISSUE THREAD

**Issue by Demigodrick** [db0/pythorhead#56]
Before I submit a PR I just wanted to check I wasnt doing something wrong and overcomplicating.

I was trying to run the following code:
```
def get_communities():
    output = lemmy.community.list(limit=10, page= 1, sort=SortType.New, type_=ListingType.Local)
    print(output)
```

But i found that the output was always ALL communities regardless of the variables, i.e listing type and sort type.

I've had a look at communities.py and changed:
```
        list_community: dict = {}

        if limit is not None:
            list_community["limit"] = limit
        if page is not None:
            list_community["page"] = page
        if sort is not None:
            list_community["sort"] = sort.value
        if type_ is not None:
            list_community["type"] = type_.value

        if data := self._requestor.api(Request.GET, "/community/list", params=list_community):
            return data["communities"]
        return []
```
to
```
        params: dict[str, Any] = {key: value for key, value in locals().items() if value is not None and key != "self"}
        return self._requestor.api(Request.GET, "/community/list", params=params)
```

Which now works when I pass the following:
```
def get_communities():
    output = lemmy.community.list(limit=10, page= 1, sort="New", type_="Local")
    print(output)
```
To circle back to the original question - is this a bug in the original code (in which case I'll submit a PR) or is it me running the query wrong?

Thanks

**Comment by db0** [comment]
The code you replaced with should work identically. I'll run some tests

**Comment by NicKoehler** [comment]
the issue is here:
```python
if type_ is not None:
    list_community["type"] = type_.value # < here
```

it needs to be changed to:
```python
if type_ is not None:
    list_community["type_"] = type_.value # see underscore was missing on the "type" string
```

or just as @Demigodrick mentioned:
```python
params: dict[str, Any] = {key: value for key, value in locals().items() if value is not None and key != "self"}
```

**Comment by db0** [comment]
Ye the latter approach is better. It's what I've been using already as its future proof.

**Comment by NicKoehler** [comment]
The only problem with using this:

```python
params: dict[str, Any] = {key: value for key, value in locals().items() if value is not None and key != "self"}
```

is that there are enums where the actual value needs to be .value or we can change the enum class to do that automatically

**Comment by db0** [comment]
Not sure what you mean by "value needs to be .value"

**Comment by NicKoehler** [comment]
> Not sure what you mean by "value needs to be .value"

I mean like this:
```python
list_community["type_"] = type_.value
```

**Comment by db0** [comment]
Fixed in #58
